lib/semantics/check-do.cc

```diff
@@ -523,7 +514,7 @@ class DoContext {
   void CheckMaskIsPure(const parser::ScalarLogicalExpr &mask) const {
     SymbolSet references{GatherSymbolsFromExpression(mask.thing.thing.value())};
     for (const Symbol *ref : references) {
-      if (IsProcedure(*ref) && !IsPureProcedure(*ref)) {
+      if (ref && IsProcedure(*ref) && !IsPureProcedure(*ref)) {
```

Comment:
Same comment here. `ref` can't be null.

Reply:
Yes, and in the case, it's easy to verify.